app/questionnaire/router.py

```diff
@@ -70,6 +70,11 @@ def get_first_incomplete_location_in_questionnaire_url(self) -> str:
 
     def get_last_location_in_questionnaire_url(self) -> str | None:
         section_key = self._get_last_complete_section_key()
+        if section_key and self.can_display_section_summary(section_key):
```

Comment:
Optional non blocker - you could use the walrus operator and nest both paths under section key existing to save two checks

Reply:
Ah cool! I'll push something up to see if I've understood correctly. Thanks 👍

Reply:
Yep - what you did was exactly what I was thinking!